openqaoa/optimizers/result.py

```diff
@@ -89,11 +90,10 @@ def __init__(
                 log.job_ids.best[0] 
                 if len(log.job_ids.best) != 0 else []
         }
-
+        
         self.most_probable_states = most_probable_bitstring(
-            cost_hamiltonian, self.get_counts(log.measurement_outcomes.best[0])
-        ) if log.measurement_outcomes.best != [] else []
-
+                cost_hamiltonian, self.get_counts(log.measurement_outcomes.best[0])
+            ) if backend.__name__ != "QAOABackendAnalyticalSimulator" and log.measurement_outcomes.best != [] else [] 
```

Comment:
If backend is already a string, why are you accessing `backend.__name__` ?

Reply:
Instead of comparing the name we could do `isintance(backend, QAOABackendAnalyticalSimulator)`

Reply:
It depends on the datatype of the backend. I think passing a string may be leaner than passing the object, but either works